# How to Increase Mobile App Revenue

Our goal is to find mobile app profiles that are more likely to attract users and to increase ad revenue for apps available on Google Play and the App Store. We are working as data analysts for this game development company.

At our company, we work on apps that are free to download and install, and our main source of revenue consists of in-app ads. Our revenue is mostly based on the amount of users; as more users engage with our ads, we are able to increase profits. By the end of our project, we can hopefully help developers understand what apps attract more users.

## Step 1: Explore the Given Data

In [52]:
from csv import reader

opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

In [53]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [54]:
print(android_header)
explore_data(android, 0, 3, True)
print('\n')

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13




In [55]:
print(ios_header)
explore_data(ios, 0, 3, True)
print('\n')

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16




In [56]:
print(android[10472])
print('\n')
print(android_header)
print('\n')
print(android[0])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


## Step 2: Clean the Given Data

However, we need to remember that there can be *errors* in the rows. According to a Google Play Store discussion, there is an error where an additional row is included and there are non-free apps. Below, we can edit our list to remove that row.

In [57]:
print(len(android))
del android[10472]
print(len(android))

10841
10840


As such, we can see that after we remove our error row, we have 10840 instead of 10841 rows.

We also need to check our data to look for *duplicate* data. For instance, is Instagram mentioned more than once in the list of apps for Android.

In [58]:
duplicate_android_apps = []
unique_android_apps = []

for app in android:
    name = app[0]
    if name in unique_android_apps:
        duplicate_android_apps.append(name)
    else:
        unique_android_apps.append(name)

print('Number of duplicate android apps:', len(duplicate_android_apps))
print('\n')
print('Example of duplicate android apps:', duplicate_android_apps[:10])

Number of duplicate android apps: 1181


Example of duplicate android apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


In [59]:
duplicate_ios_apps = []
unique_ios_apps = []

for app in ios:
    name = app[0]
    if name in unique_ios_apps:
        duplicate_ios_apps.append(name)
    else:
        unique_ios_apps.append(name)

print('Number of duplicate ios apps:', len(duplicate_ios_apps))
print('\n')
print('Example of duplicate ios apps:', duplicate_ios_apps[:10])

Number of duplicate ios apps: 0


Example of duplicate ios apps: []


Our data has showed us that there are **10840 duplicate Android apps** and **0 duplicate iOS apps**.

Now, we are going to create a dictionary `reviews_max` to find & remove the duplicate rows. We are doing the following:
- we create a dictionary where the `name` of the app is the kay and the `n_reviews` is the value
- the dictionary should provide 9659 entries

In [60]:
reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

print(len(reviews_max))

9659


Now, we are going to craft a list of applications to add to the cleaned list without any dupplicates. Both lists `android_clean` and `already_added` will reflect 9659 apps.

In [61]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if (n_reviews == reviews_max[name]) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)

print(len(already_added))

9659


In [62]:
explore_data(android_clean, 0, 3, False)
print("Number of rows:", len(android_clean))

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659


As we explore our data, we can see the same result as we'd previously arrived at. Our data shows us that there are 9,659 rows in our newly-cleaned dataset for android apps.

Another important step in cleaning our data is making sure that we are analyzing *English Apps*, or apps that are in English for American advertisement companies.

As such, we are going to create `eng(string)` a function that checks whether there are non-English characters.

In [63]:
def eng(string):
    for character in string:
        if ord(character) > 127:
            return False
        
    return True

print(eng('Instagram'))
print(eng('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(eng('Docs To Go™ Free Office Suite'))
print(eng('Instachat 😜'))

True
False
False
False


Our function definining `eng(string)` was successful in identifying the apps 'Instagram' as English and '爱奇艺PPS -《欢乐颂2》电视剧热播'
as non-English.

However, the other two apps,'Docs To Go™ Free Office Suite' and 'Instachat 😜', had issues in being deemed "English apps". This is because of emojis and characters like "™" that fall outside the ASCII range. We can see evidence of that below.


In [64]:
print(ord('™'))
print(ord('😜'))

8482
128540


Above, we can see corresponding numbers above the ASCII range of 127 that we used for our original `eng(string)` function. We're going to craft a new function to check whether an app is English.

In [65]:
def eng(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True

print(eng('Docs To Go™ Free Office Suite'))
print(eng('Instachat 😜'))

True
True


Now that we've designed a semi-functional way to exclude English apps, we're going to make two lists, `android_english` and `ios_english`, that define applications in both application stores that are English.

In [66]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if eng(name):
        android_english.append(app)
        
for app in ios:
    name = app[1]
    if eng(name):
        ios_english.append(app)
        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)
        

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

According to our data-cleaning process, the number of English Android apps is **9,614** and the number of English iOS apps is **6,183**. 

Now, we need to make our final lists, where the apps are both *English* and *Free*. These lists will be called `android_final` and `ios_final`.

In [67]:
android_final = []
ios_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)

for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)

print(len(android_final))
print(len(ios_final))

8864
3222


Thus far, we've done the following to clean our data:
- removed inaccurate data
- removed duplicate app entries
- removed non-English apps
- isolated the free apps

## Step 3: Product Strategy

To find apps that are successful and attract the most ad revenue, we are going to launch the app on the Google Play Store *first* and the App Store after *proof-of-concept* with Android. This can be described in three unique steps:
1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

After inspecting both data sets, we need to find the most common genres in each market. This is the 'Genre' and 'Category' columns for Android and 'prime_genre' column for iOS.

We'll build frequency tables for these columns. To do so, we'll build two functions:
1. `freq_table()`: generate *frequency tables that show percentages*
2. `display_table()`: display *percentages in a descending order*

In [68]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        table_percentages[key] = (table[key] / total) * 100
    
    return table_percentages
    

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

We're going to use our new functions to display frequency table of the columns `prime_genre`, `Genres`, and `Category`.

In [69]:
display_table(ios_final, -5) #"prime_genre"

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Our data analysis shows the following:
The most popular category is "Games" at approximately 58.2%. Other notable runners-up include "Entertainment" at 7.9%, "Photo & Video" at 5.0%, and "Education" at 3.7%.
The least popular category is "Catalogs" at approximately 0.1%, "Medical" at 0.2%, "Navigation" at 0.2%, and "Book" at 0.4%.

This information helps us realize that more than half of the apps on the app store that are *free* and *English* are related to **fun**(Games, Entertainment, Photo&Video, etc).

However, this does not imply that most people use the App Store to *download* games; instead, this is just information about offered-products–not *usage* or *demand*.


In [70]:
display_table(android_final, 1) #"Category"

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

Most apps on the Google Play Store seem to be designed for family, game, tools, business, and lifestyle. This is vastly different from the App Store, whose product line is mostly comrpised of games for fun and entertainment.

At a closer look, the title of "family" is misleading because a lot of the products offered here are games for younger children. We can see that at least 28% of products on the Google Play Store can be attributed to fun products like games.

In [71]:
display_table(android_final, -4) #"Genres"

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

This display for Genres contains way more rows/categories. If we want to simplify our analysis, we may prefer the `Category` table because of the more narrow data-set. 

As of now, we've found that roughly *60%* of applications on the iOS App Store are dominated by games, whereas roughly *30%* of applications on the Google Play Store are dominated by games. There is more balance and variety in applications on the Google App Store, as seen by the smaller range in the `Category` column and the larger range in the `prime_genre` column.

We've also realized a strong limitation in our current data-set, which indicates the *proportion of apps based on genre* instead of the *number of users* or *demand based on genre*


## Step 4: App Profile Recommendations 

### App Profile Recommendation for iOS App Store

In [74]:
genres_ios = freq_table(ios_final, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total/len_genre
    print(genre, ':', avg_n_ratings)


Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


According to our data, we can see that "Navigation" has the highest average number of users at 86,090. However, we can further break this down to see which apps are in this list.

In [76]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ":", app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


This breakdown shows Google Maps and Waze make up the largest contribution to the popularity of 'Navigation' among users.

Within this breakdown, we can come to a couple of guidelines that may shape the profile we choose:
1. The market for 'Games' and 'Entertainment' is oversaturated. There are so many products and different games that serve as sources of entertainment–this makes it really hard to find competitive advantages and ways to attract audiences to our company's app if we decide to make a game.
2. Categories such as 'Navigation' and 'Reference' are really popular because of key big players. There are many products in those areas too, but they're relatively uncompetitive given use of apps such as Google Maps and Waze instead of unpopular services such as Railway Route Search.
3. We want to design an app that requires the least amount of domain knowledge. This means that we don't want to spend the capital required to hire financial liasons for building baking apps or legal professors for a LSAT-preparation app.
4. Apps that are most profitable for add-revenue are apps that people spend large amounts of time on. This means apps that are weather-oriented may have large users, but users aren't checking the weather long enough to engage with advertisements.
5. The best possible outcome is for our company to build an app that requires minimal consumer education. Building an app that requires massive advertisement and change in behavior may be beyond the scope of a small app development company, seeking to make a profit. 

Based on these initial thoughts, we recommend the following for the iOS App Store: 

**The company should build an application that creates an all-in-one calendar for its users.** Here, users can plan out their day, write down reminders, and categorize action items.
- Fights oversaturation of games in the app store by introducing a new product that is not based on entertainment or fun.
- Limited competition with Notion and Reminders - those are popular, but no one service dominates the entire market.
- Keeps users engaged, as they'll regularly check the app to update progress and track check-lists

### App Profile Recommendation for Google Play Store

In [78]:
display_table(android_final, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


According to our data more than 15.7% of apps have 1,000,000+ downloads and 0.01% of apps have 0 downloads.

NOTE: this frequency table's data is imprecise because the installs are divided in ranges. 

Now, we're going to use our data to find the average number of installs per app genre for the Google Play dataset.

In [85]:
categories_android = freq_table(android_final, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

According to our data, apps that are a part of the "Communication" category with more than 38,456,119 downloads on average. 

We can further break-down this category to see what apps dominate this category.

In [86]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])


WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

In [94]:
for app in android_final:
    if app[1] == 'TOOLS' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

n_of_tool_based_apps = 0

for app in android_final:
    if app[1] == 'TOOLS':
        n_of_tool_based_apps += 1

print('\n')
print('Number of Tool-Based Apps: ', n_of_tool_based_apps)

Moto Display : 10,000,000+
Motorola Alert : 50,000,000+
Motorola Assist : 50,000,000+
Moto Suggestions ™ : 1,000,000+
Moto Voice : 10,000,000+
myMetro : 10,000,000+
File Manager : 50,000,000+
My Telcel : 50,000,000+
Calculator - free calculator, multi calculator app : 10,000,000+
ASUS Sound Recorder : 10,000,000+
iWnn IME for Nexus : 5,000,000+
Samsung Max - Data Savings & Privacy Protection : 10,000,000+
Android TV Remote Service : 1,000,000+
ZenUI Help : 10,000,000+
ZenUI Keyboard – Emoji, Theme : 10,000,000+
Files Go by Google: Free up space on your phone : 10,000,000+
SD card backup : 1,000,000+
Nokia mobile support : 5,000,000+
File Manager -- Take Command of Your Files Easily : 10,000,000+
Clear : 10,000,000+
Phone : 10,000,000+
HTC Lock Screen : 10,000,000+
AT&T Smart Wi-Fi : 10,000,000+
Google app for Android TV : 10,000,000+
Sound Recorder: Recorder & Voice Changer Free : 10,000,000+
Remote Link (PC Remote) : 10,000,000+
HTC Sense Input : 10,000,000+
App vault : 10,000,000+
To

Through our analysis, we can make two claims:
1. The communication genre is comprised of really big players like WhatsApp and Messenger, which would make it very hard for  substitutes and new entrants to compete.
2. Tools are extremely popular. Even though there are *750 different tool-based apps*, there are still more than 1,000,000+ downloads for each of these tools.

This confirms that if we introduce the product we designed for the iOS App Store, there is a chance that it can succeed in the Google Play Store. In fact, it may succeed even more—because there is more balance in the Google Play Store and less saturation of games/entertainment apps. This provides a track-record of successful tools and lifestyle-based apps that enhance users' daily lives.

As such, we'd recommend that the company  build an application that creates an all-in-one calendar for its users